cmd/dagger/session.go

```diff
@@ -39,6 +40,10 @@ type connectParams struct {
 }
 
 func EngineSession(cmd *cobra.Command, args []string) error {
+	// discard SIGPIPE, which can happen when stdout or stderr are closed
+	// (possibly from the spawning process going away)
+	signal.Notify(make(chan os.Signal, 1), syscall.SIGPIPE)
```

Comment:
cc @jedevc for continuity - ended up having to add an even harder fix for this, since it was still happening, only this time for stderr. Seems fine to just drop all SIGPIPEs.

Reply:
This makes sense and sgtm; I was reading the linux docs about `SIGPIPE` and got kind of confused about what the end effect would be here with the go runtime, but the go docs answered it clearly: https://pkg.go.dev/os/signal#hdr-SIGPIPE

Maybe worth including that link in the comment for any future confused individuals

Reply:
SGTM - writing to stdout failing with an error seems like the expected behavior for this.

@vito if you were seeing this, then that might have been a symptom of something else - why was stderr being closed in the first place? It's likely fine, but the only reason I noticed this in the first place was because the `SIGPIPE` behavior was the only thing preventing an infinite telemetry drain.

Reply:
> @vito if you were seeing this, then that might have been a symptom of something else - why was stderr being closed in the first place? It's likely fine, but the only reason I noticed this in the first place was because the `SIGPIPE` behavior was the only thing preventing an infinite telemetry drain.

The OS closed the pipe because the spawning process does `cmd.StdoutPipe` and then exits. So nothing actually called `.Close()`, the pipe just broke on its own, and then the `dagger session` process got `SIGPIPE` when it tried to write to it.

@sipsma Yeah good point, I ended up on that same doc, and it relates to the above question, so worth a link indeed. Added!